homeassistant/components/lg_thinq/water_heater.py

```diff
@@ -0,0 +1,204 @@
+"""Support for waterheater entities."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from thinqconnect import DeviceType
+from thinqconnect.integration import ExtendedProperty
+
+from homeassistant.components.water_heater import (
+    ATTR_OPERATION_MODE,
+    STATE_ECO,
+    STATE_HEAT_PUMP,
+    STATE_OFF,
+    STATE_PERFORMANCE,
+    WaterHeaterEntity,
+    WaterHeaterEntityDescription,
+    WaterHeaterEntityFeature,
+)
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import ThinqConfigEntry
+from .coordinator import DeviceDataUpdateCoordinator
+from .entity import ThinQEntity
+
+DEVICE_TYPE_WH_MAP: dict[DeviceType, WaterHeaterEntityDescription] = {
+    DeviceType.WATER_HEATER: WaterHeaterEntityDescription(
+        key=ExtendedProperty.WATER_HEATER,
+        name=None,
+        translation_key=ExtendedProperty.WATER_HEATER,
+    ),
+    DeviceType.SYSTEM_BOILER: WaterHeaterEntityDescription(
+        key=ExtendedProperty.WATER_BOILER,
+        name=None,
+        translation_key=ExtendedProperty.WATER_BOILER,
+    ),
+}
+
+# Mapping between device and HA operation modes
+OP_STATE_TO_HA = {
+    "auto": STATE_ECO,  # Auto
+    "heat_pump": STATE_HEAT_PUMP,  # Heat pump
+    "turbo": STATE_PERFORMANCE,  # Turbo
+    "vacation": STATE_OFF,  # Vacation
+}
```

Comment:
```suggestion
    "auto": STATE_ECO,
    "heat_pump": STATE_HEAT_PUMP,
    "turbo": STATE_PERFORMANCE,
    "vacation": STATE_OFF,
}
```
the code is already self-explanatory